app/src/main/java/doit/study/droid/quiz/QuizPageFragment.kt

```diff
@@ -0,0 +1,308 @@
+package doit.study.droid.quiz
+
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.*
+import android.widget.*
+import androidx.annotation.StringRes
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProviders
+import androidx.navigation.fragment.findNavController
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import doit.study.droid.R
+import doit.study.droid.app.BaseApp
+import doit.study.droid.settings.SettingsFragment
+import doit.study.droid.utils.AnalyticsData
+import doit.study.droid.utils.Sound
+import doit.study.droid.utils.Views
+import timber.log.Timber
+import javax.inject.Inject
+import kotlin.random.Random
+
+
+class QuizPageFragment: Fragment(){
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: QuizPageViewModel
+    private lateinit var viewModelMain: QuizMainViewModel
+    private val pagePosition: Int by lazy { arguments!!.getInt(ARG_POSITION_IN_QUIZ_KEY, 0) }
+    private lateinit var commitButton: FloatingActionButton
+    private lateinit var thumpUpButton: ImageButton
+    private lateinit var thumpDownButton: ImageButton
+    private var toast: Toast? = null
+    private val sound: Sound by lazy { Sound.getInstance(activity?.applicationContext) }
+    private lateinit var question: TextView
+    private lateinit var answerContainer: LinearLayout
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        BaseApp.dagger.inject(this)
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+        viewModel = ViewModelProviders
+                .of(requireParentFragment(), viewModelFactory)
+                .get(pagePosition.toString(), QuizPageViewModel::class.java)
+
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        commitButton = view.findViewById(R.id.commit_button)
+        thumpUpButton = view.findViewById(R.id.thump_up_button)
+        thumpDownButton = view.findViewById(R.id.thump_down_button)
+        question = view.findViewById(R.id.question)
+        answerContainer = view.findViewById(R.id.answers)
```

Comment:
Kotlin synthetics may be used here as well

Reply:
It's a stale code that issue was fixed in master